dbms/src/Flash/Executor/PipelineExecutorStatus.cpp

```diff
@@ -100,8 +100,8 @@ void PipelineExecutorStatus::onEventFinish() noexcept
         if (0 == active_event_count)
         {
             // It is not expected for a query to be finished more than one time.
-            bool expect_origin = false;
-            RUNTIME_ASSERT(is_finished.compare_exchange_strong(expect_origin, true));
+            RUNTIME_ASSERT(!is_finished);
+            is_finished = true;
```

Comment:
This is not thread safe, is it expected?

Reply:
yes